packages/react/src/CTAForm/CTAForm.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react'
+import clsx from 'clsx'
+import {FormControl, Button, ButtonProps, TextInput, TextInputProps, Checkbox} from '../'
+
+import styles from './CTAForm.module.css'
+
+type CTAFormProps = {
+  inputLabel: TextInputProps['leadingText']
+  inputType?: TextInputProps['type']
+  placeholder?: TextInputProps['placeholder']
+  buttonLabel: ButtonProps<'button'>['children']
+  checkboxLabel: React.ReactNode[] | React.ReactNode | string
+} & React.HTMLAttributes<HTMLFormElement>
```

Comment:
This is quite a closed API. Was there a reason for not going with a composable API here? I can't find the API exploration to see the wider conversation.

Reply:
E.g. How would I use this form in either controlled or uncontrolled modes? I can't pass a `ref`, or at least it's not clear to me right now how I would grab this from the form submission event. Also the `method` and `action` attributes are forwarded from what I can see, but wondering if something like this might make this clearer.

```jsx
  <CTAForm method="post" action="...">
    <CTAForm.TextInput ref={inputRef} />
    {needsCheckbox && <CTAForm.Confirm label={checkboxLabel} />}
  <CTAForm>
``` 
In the current format, I'm not sure how I can conditionally render the checkbox field, which is important because it's not always required.